### PR TITLE
Creating test fixture to reset state of TF Keras graph before each test

### PR DESCRIPTION
**Pull request by gabrielspmoreira**
I have observed that some unit tests that run successfully when executed individually were failing when multiple tests are run.
@oliverholworthy suggested [here](https://github.com/NVIDIA-Merlin/models/pull/511) using this Pytest fixture to clear the session before every test. I tested that approach and was able to uncomment some test asserts that check for metric names of default Keras metric (e.g. auc, precision, recall) which before were given different names (e.g. auc, auc_2, auc_3) depending on tests executed before.

```python
@pytest.fixture(autouse=True)
def tf_clear_session():
    tf.keras.backend.clear_session()
```
<!--

### CODE REVIEW

**Review by benfred — APPROVED** [review]
looks good! nice suggestion @oliverholworthy
